pkg/scheduler/framework/session.go

```diff
@@ -320,6 +322,29 @@ func (ssn *Session) Backoff(job *api.JobInfo, event arbcorev1.Event, reason stri
 	return nil
 }
 
+// UpdatePodCondition will update pod status of all pending tasks of a job
+func (ssn *Session) UpdatePodCondition(job *api.JobInfo) error {
+	jobErrorMsg := job.FitError()
+	// All pods Conditions need to be updated
+	for _, tasksMap := range job.TaskStatusIndex {
```

Comment:
we can not update all Pods/Tasks's status, some task/pod maybe already in running state.

Reply:
At the beginning, I thought to update podCondition and status only for pods failed in scheduling. After the real test, I notice that's not that friendly to autoscaler.. because failed pods can fit into existing resource. That's why I change to update all pods. 

Since you already mentioned we can work on the normal case firstly and then reconsider how to interact with cluster autoscaler, I will change it back.

Reply:
> After the real test, I notice that's not that friendly to autoscaler.. because failed pods can fit into existing resource

For this case, we only need to update pod condition for `allocated` pod; refer to what we had done in `Backoff` :)